packages/communication-react/package.json

```
@@ -45,6 +43,8 @@
     "@fluentui/react-hooks": "~8.3.8"
   },
   "peerDependencies": {
+    "@azure/communication-calling": "1.3.2",
```

if we do this, presumably when someone does an npm install they also have to do `npm i @azure/communication-calling  @azure/communication-chat`? If so we'd need to update documentation along side this

Given we limit to a single version it's odd to have a peer dependency of a fixed version (although I do see the need). React hooks can detect if there's multiple versions of react and throws a big error - could that be a solution instead?

---

If only we didn't ask our clients to use `--legacy-peer-deps` :( (I'm looking at your `react-northstar`).

Without this flag, the default [NPM behavior since v7 is to auto-install peer-dependencies](https://stackoverflow.com/questions/66239691) so that they wouldn't necessarily have to install our peer dependencies explicitly.

That said, peer dependencies are the way to go here:

* We do plan to move to a minimum-supported-version for these dependencies. At least allow patch version updates for chat/calling without needing to release a new communication-react stable version.
* Detecting multiple versions of a package and complaining -- if this is the path we take, it'd have to be the package that dislikes multiple versions (i.e., calling) that does that. Also, it feels wrong to solve this problem at runtime in the package, when NPM dependency management is already intended to solve it.

Perhaps `@azure/communication-calling` _should_ detect multiple versions and complain, especially if they decide it's a BAD thing to happen. I'd say that it should be in addition to proper dependency management though.

---

Can we have these allow minor versions today? 
```suggestion
    "@azure/communication-calling": ">=1.3.0 <1.4.0",
```

---

(I did file an issue with n* to support v17+ https://github.com/microsoft/fluentui/issues/21082)

---

(perhaps we could even hoist or bundle it out of the fluent repo: https://github.com/microsoft/fluentui/tree/master/packages/fluentui/react-northstar/src/components/Chat)

---

Replying to the suggestion for this PR: I don't want to change the actual version supported in any way in this PR (just move the single pinned version to peer deps).

We can start supporting minor versions in the future, but let's keep that a separate PR / discussion. The main concern there is knowing what our support plan is -- folks might start using latest calling before we've integrated it / tested with our own stable release.